src/tests/Autosuggest.js

```diff
@@ -412,6 +426,129 @@ describe('Autosuggest', function() {
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbStrings}
+                     onSuggestionFocused={onSuggestionFocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+      it('should call onSuggestionFocused when suggestion focused using mouse', function() {
+        mouseOverFromInputToSuggestion(1);
+        expect(onSuggestionFocused).toBeCalledWith('Mordialloc');
+      });
+    });
+
+    describe('Keyboard interactions', function() {
+      it('should call onSuggestionFocused when suggestion focused using Up/Down keys', function() {
+        clickDown();
+        expect(onSuggestionFocused).toBeCalledWith('Mill Park');
+      });
+
+      it('should not call onSuggestionFocused when Up/Down keys are pressed after ESC is pressed', function() {
+        clickDown();
+        clickEscape();
+        onSuggestionFocused.mockClear();
+        clickDown();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+
+      it('should not call onSuggestionFocused when first suggestion is focused and Up key is clicked', function() {
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickUp();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+
+      it('should not call onSuggestionFocused when last suggestion is focused and Down key is clicked', function() {
+        clickDown();
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickDown();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+    });
+  });
+
+  describe('onSuggestionUnfocused', function() {
+    beforeEach(function() {
+      onSuggestionUnfocused.mockClear();
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbObjects}
+                     suggestionRenderer={renderSuburbObject}
+                     suggestionValue={getSuburbObjectValue}
+                     onSuggestionUnfocused={onSuggestionUnfocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+      it('should call onSuggestionUnfocused when suggestion unfocused using mouse', function() {
+        mouseOverFromInputToSuggestion(0);
+        mouseOverBetweenSuggestions(0, 1);
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should call onSuggestionUnfocused when suggestion selected with mouse', function() {
+        mouseOverFromInputToSuggestion(0);
+        mouseDownSuggestion(0);
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+    });
+
+    describe('Keyboard interactions', function() {
+      it('should call onSuggestionUnfocused when suggestion unfocused using Up/Down keys', function() {
+        clickDown();
+        clickDown();
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
```

Comment:
I'd also be inclined to check that `onSuggestionUnfocused` isn't called after the first `clickDown();`. If that was the case, this test would still pass.